### ISSUE TITLE
[Bug]: List of apps disappears when updating

### ISSUE THREAD

**Issue by paulchen** [nextcloud/server#42747]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When updating the apps in the web UI, the list of apps disappears for some reason.

### Steps to reproduce

1. Wait for an app update to be available.
2. In the web UI, log in using an admin user and navigate to the list of installed apps ("Your apps").
3. Click the "Update" button.
4. As soon as the password overlay is displayed, the list of apps disappears.
5. When closing the password overlay instead of confirming it, an error is shown ("An error occurred during the request. Unable to proceed."). I think this error is kind of confusing, but I don't know whether this can be considered a bug as well.
6. Click on "Your apps". Nothing happens.
7. Click on "Active apps" and again on "Your apps". The list of apps will now be shown again.

https://github.com/nextcloud/server/assets/404840/b7d809c5-fbdd-4736-8b69-7acf37ad410c

The list of apps not being shown does not have an influence on the update process itself. The apps can be updated as expected.

### Expected behavior

The list of apps does not disappear in step 4.

### Installation method

Community Manual installation with Archive

### Nextcloud Server version

28

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.2

### Web server

Nginx

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Upgraded to a MAJOR version (ex. 22 to 23)

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.huntu.at"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "default_phone_region": "AT", 
        "dbtype": "mysql",
        "version": "28.0.1.1",
        "overwrite.cli.url": "https:\/\/cloud.huntu.at",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "maintenance": false,
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "smtp",  
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauth": 1,
        "mail_smtpsecure": "tls",
        "mail_smtpport": "587",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN", 
        "theme": "",
        "loglevel": 0,
        "mysql.utf8mb4": true,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "preview_max_memory": -1,
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "timeout": 1.5
        },
        "updater.release.channel": "stable",
        "app_install_overwrite": [
            "calendar"
        ],
        "enable_previews": false
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 2.20.0
  - admin_audit: 1.18.0
  - calendar: 4.6.3
  - circles: 28.0.0-dev
  - cloud_federation_api: 1.11.0
  - contacts: 5.5.1
  - contactsinteraction: 1.9.0
  - dashboard: 7.8.0
  - dav: 1.29.1
  - deck: 1.12.0
  - federatedfilesharing: 1.18.0
  - files: 2.0.0
  - files_pdfviewer: 2.9.0
  - files_reminders: 1.1.0
  - files_sharing: 1.20.0
  - files_trashbin: 1.18.0
  - files_versions: 1.21.0
  - firstrunwizard: 2.17.0
  - forms: 4.0.0
  - logreader: 2.13.0
  - lookup_server_connector: 1.16.0   
  - nextcloud_announcements: 1.17.0   
  - notifications: 2.16.0
  - oauth2: 1.16.3
  - password_policy: 1.18.0
  - photos: 2.4.0
  - privacy: 1.12.0
  - provisioning_api: 1.18.0
  - recommendations: 2.0.0
  - related_resources: 1.3.0
  - richdocuments: 8.3.0
  - serverinfo: 1.18.0
  - settings: 1.10.1
  - sharebymail: 1.18.0
  - survey_client: 1.16.0
  - systemtags: 1.18.0
  - tasks: 0.15.0
  - text: 3.9.1
  - theming: 2.3.0
  - twofactor_backupcodes: 1.17.0
  - updatenotification: 1.18.0
  - user_status: 1.8.1
  - viewer: 2.2.0
  - weather_status: 1.8.0
  - workflowengine: 2.10.0
Disabled:
  - bruteforcesettings: 2.8.0
  - checksum: 1.2.2 (installed 1.2.2) 
  - collectives: 2.9.2 (installed 2.9.2)
  - comments: 1.18.0 (installed 1.5.0)
  - encryption: 2.16.0
  - federation: 1.18.0 (installed 1.6.0)
  - files_external: 1.20.0
  - files_markdown: 2.4.1 (installed 2.4.1)
  - notes: 4.9.2 (installed 4.9.2)
  - polls: 6.0.1 (installed 6.0.1)
  - quicknotes: 0.8.10 (installed 0.8.10)
  - support: 1.11.0 (installed 1.0.0)
  - suspicious_login: 6.0.0
  - twofactor_totp: 10.0.0-beta.2
  - user_ldap: 1.19.0
```

### Nextcloud Signing status

```shell
No errors have been found.
```

**Comment by joshtrichards** [comment]
Hi @paulchen - Are you using a WebKit based browser by chance? There was a similar, but not example the same report in #42249 your situation reminds me of.

**Comment by paulchen** [comment]
> Hi @paulchen - Are you using a WebKit based browser by chance?

Yes, its Chromium (`Version 120.0.6099.216 (Offizieller Build) Arch Linux (64-Bit)`). I should have mentioned that above.

However, since I have updated the updated the app in the meantime, I need to wait for the next app update to check whether the problem also occurs when using Firefox.

> There was a similar, but not exactly the same report, in #42249 that your situation reminds me of.

I can't tell whether that one may be related. I did search for existing issues, but I was unable to narrow down the search well enough to find issues that are really similar.

**Comment by paulchen** [comment]
> > Hi @paulchen - Are you using a WebKit based browser by chance?
> 
> Yes, its Chromium (`Version 120.0.6099.216 (Offizieller Build) Arch Linux (64-Bit)`). I should have mentioned that above.
> 
> However, since I have updated the updated the app in the meantime, I need to wait for the next app update to check whether the problem also occurs when using Firefox.

As today an update for the Calendar app from 4.6.3 to 4.6.4 became available, I was able to check whether this problem is present on Firefox (version 121.0.1 (64-bit) on Arch Linux): It is. The behaviour is the same as on Chrome.

What I also noticed: If you don't confirm your password, but close the overlay, navigate to "Active apps" and back to "Your apps", and finally click on the "Update" button once more, the problem does not occur. After hitting F5 or using the browser's reload button, the problem can be reproduced again. There is no difference in this behaviour between Firefox and Chrome.


https://github.com/nextcloud/server/assets/404840/98de7af0-efce-4e93-b108-3f190d7b5e66

**Comment by joshtrichards** [comment]
Not seeing this behavior today. Are you still seeing it?

**Comment by paulchen** [comment]
> Not seeing this behavior today. Are you still seeing it?

I just checked it on Nextcloud 30.0.6 and did not observe that behaviour either. Hence I close this issue now.

Thanks to whoever fixed it. :)
